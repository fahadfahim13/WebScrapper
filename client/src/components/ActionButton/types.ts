import { ReactElement } from "react";
import {SvgIconProps} from "@mui/material";

import {BUTTON_COLORS, BUTTON_TYPES} from "shared/types";

export interface Props {
    variant?: `${BUTTON_TYPES}` | undefined;
    startIcon?: ReactElement<SvgIconProps>;
    value: string;
    color?: `${ BUTTON_COLORS }` | undefined;
    onClick: () => void;
}
