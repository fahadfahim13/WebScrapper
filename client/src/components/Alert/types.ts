import { SyntheticEvent, MouseEvent } from "react";
import { VERTICAL, HORIZONTAL } from "shared/types";

export interface Props {
    open: boolean;
    autoHideDuration?: number;
    message: string;
    vertical?: `${VERTICAL}`;
    horizontal?: `${HORIZONTAL}`;
    onClose: (event: SyntheticEvent | MouseEvent) => void;
}