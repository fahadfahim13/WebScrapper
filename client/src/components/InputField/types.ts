import { ChangeEvent } from "react";
import {InputTypes} from "store/FormInputs/types";

export interface Props {
    id?: string;
    name: InputTypes;
    value: string;
    type?: string;
    required?: boolean;
    label?: string;
    defaultValue?: string;
    onChange: (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
    multiline?: boolean;
    maxRows?: number;
    width?: string;
}