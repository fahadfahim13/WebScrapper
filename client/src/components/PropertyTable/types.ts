import { PropertyDetails } from "shared/types";

export interface Props {
    tableBody?: PropertyDetails[];
}