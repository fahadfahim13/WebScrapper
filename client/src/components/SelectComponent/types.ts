import { SelectChangeEvent } from "@mui/material";

export interface Option{
    id: number;
    value: string;
    text: string;
}

export interface Props {
    options: Option[];
    handleChange: (e: SelectChangeEvent) => void;
    value: Option;
    label?: string;
}