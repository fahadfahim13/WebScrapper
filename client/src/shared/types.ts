export enum BUTTON_TYPES {
    OUTLINED = "outlined",
    TEXT = "text",
    CONTAINED = "contained",
}

export enum BUTTON_COLORS {
    INHERIT = "inherit",
    PRIMARY = "primary",
    SECONDARY = "secondary",
    SUCCESS = "success",
    ERROR =  "error",
    INFO = "info",
    WARNING = "warning"
}

export enum VERTICAL {
    TOP = "top",
    BOTTOM="bottom",
}

export enum HORIZONTAL {
    CENTER = "center",
    LEFT = "left",
    RIGHT = "right",
}