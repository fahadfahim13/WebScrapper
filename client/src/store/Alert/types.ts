export interface State {
  id: string | number;
  message: string;
  open: boolean;
}
