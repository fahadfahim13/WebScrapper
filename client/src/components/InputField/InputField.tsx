import React from 'react';
import { TextField } from '@mui/material';

import {Props} from './types';

const InputField = (props: Props) => {
    const {required=false, name='default', id='Default', type='text',
        label='Default', onChange, value='',
        multiline=false, maxRows=1, width='100%'} = props
    return (
        <TextField
            required={required}
            id={id}
            label={label}
            name={name}
            type={type}
            value={value}
            onChange={onChange}
            multiline={multiline}
            maxRows={maxRows}
            style={{ width: width}}
        />
    );
};

export default InputField;