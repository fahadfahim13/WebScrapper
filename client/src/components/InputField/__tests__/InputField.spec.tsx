import React from 'react';
import {render, RenderResult} from '@testing-library/react';
import InputField from 'components/InputField';
import {Props} from '../types';
import {INPUT_NAMES} from "uicontainers/FormInputs/types";

let documentBody: RenderResult;

describe("<InputField />", () => {
    const props: Props = {
        id: '1',
        name: INPUT_NAMES.NAME,
        value: 'SAVE',
        type: 'text',
        required: false,
        label: INPUT_NAMES.NAME,
        defaultValue: '',
        multiline: true,
        maxRows: 3,
        width: '100%',
        onChange: (e) => {
            console.log(e.target.value)
        }
    }
    beforeEach(() => {
        documentBody = render(<InputField {...props} />)
    });
    it('matches snapshot', () => {
        const {baseElement} = documentBody;
        expect(baseElement).toMatchSnapshot();
    })

});


// jest.mock('@mui/material', () => ({
//     ...jest.requireActual('@mui/material'),
//     TextField: 'TextField',
// }));
//
//
// describe('InputField component testing', () => {
//     describe('Input Field Testing', () => {
//         it('should match Input field snapshot', () => {
//             const navigationTree = create(<InputField />).toJSON();
//             expect(navigationTree).toMatchSnapshot();
//         });
//     });
// });
