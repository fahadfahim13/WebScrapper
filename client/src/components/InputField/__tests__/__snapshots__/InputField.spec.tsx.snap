// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      style="width: 100%;"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="1"
        id="1-label"
      >
        name
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id="1"
          name="name"
          style="height: 0px; overflow: hidden;"
        >
          SAVE
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1z7n62"
          >
            <span>
              name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</body>
`;
